code/controllers/Processes/lighting.dm

```diff
@@ -1,62 +1,60 @@
+// Solves problems with lighting updates lagging shit
+// Max constraints on number of updates per doWork():
+#define MAX_LIGHT_UPDATES_PER_WORK		100
+#define MAX_OVERLAY_UPDATES_PER_WORK	1000
+
 /datum/controller/process/lighting
 	schedule_interval = LIGHTING_INTERVAL
 
-	// Solves problems with lighting updates lagging shit
-	// Max constraints on number of updates per doWork():
-	var/const/MAX_LIGHT_UPDATES_PER_WORK=100
-	var/const/MAX_OVERLAY_UPDATES_PER_WORK=1000 // idfk
-
-	// Counters
-	var/light_updates=0
-	var/overlay_updates=0
-
 /datum/controller/process/lighting/setup()
 	name = "lighting"
 
 	create_lighting_overlays()
 
 /datum/controller/process/lighting/doWork()
+	// Counters
+	var/light_updates	= 0
+	var/overlay_updates	= 0
+
 	var/list/lighting_update_lights_old = lighting_update_lights //We use a different list so any additions to the update lists during a delay from scheck() don't cause things to be cut from the list without being updated.
-	lighting_update_lights = null //Nulling it first because of http://www.byond.com/forum/?post=1854520
 	lighting_update_lights = list()
-	light_updates=0
 	for(var/datum/light_source/L in lighting_update_lights_old)
-		if(L)
+		if(light_updates >= MAX_LIGHT_UPDATES_PER_WORK)
+			lighting_update_lights += lighting_update_lights_old
```

Comment:
I realised this later, I need to only add everything that's not been processed, this would fuck up the entire lighting process.